tests/Unit/Domain/CoordinateMaps/Test_ProductMaps.cpp

```diff
@@ -4,18 +4,58 @@
 #include "tests/Unit/TestingFramework.hpp"
 
 #include <array>
+#include <boost/optional.hpp>
 #include <cstddef>
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
 #include "Domain/CoordinateMaps/Affine.hpp"
 #include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/Wedge2D.hpp"
 #include "Domain/LogicalCoordinates.hpp"
 #include "Domain/Mesh.hpp"
+#include "Domain/OrientationMap.hpp"
 #include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
+namespace {
+void test_product_two_maps_fail() {
+  const CoordinateMaps::Affine affine(-1.0, 1.0, 3.0, 4.0);
+  const CoordinateMaps::Wedge2D wedge(0.2, 4.0, 0.0, 1.0, OrientationMap<2>{},
+                                      true);
+  {
+    const CoordinateMaps::ProductOf2Maps<CoordinateMaps::Affine,
+                                         CoordinateMaps::Wedge2D>
+        map(affine, wedge);
+    // Should fail wedge map
+    const std::array<double, 3> mapped_point1{{3.5, -1.0, -1.0}};
+
+    // Should be ok
+    const std::array<double, 3> mapped_point2{{3.5, 3.0, -1.0}};
+
+    CHECK(not map.inverse(mapped_point1));
```

Comment:
Why not CHECK_FALSE and CHECK (to avoid `not`s?)

Reply:
The `not s` casts to a bool, which is why we want it.  Equivalently we could have `static_cast<bool>(s)` instead.  (If you don't cast to bool, you get a compiler error about boost::optional not having stream operators.  I don't think we care about writing stream operators for boost::optional just for a test where all you care about is whether the boost::optional is true or false).  I'm ok with changing to static_cast if people like it better.  Thoughts?

Reply:
I'm changing to `static_cast<bool>` because that makes the logic clearer and explicitly says what I am doing.